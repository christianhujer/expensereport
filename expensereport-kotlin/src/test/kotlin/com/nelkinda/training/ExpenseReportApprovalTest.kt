package com.nelkinda.training

import org.approvaltests.ApprovalUtilities
import org.approvaltests.Approvals
import org.junit.jupiter.api.Test

class ExpenseReportApprovalTest {

    @Test
    fun test() {
        val output = ApprovalUtilities().writeSystemOutToStringBuffer()

        val expense = Expense()
        expense.type = ExpenseType.BREAKFAST
        expense.amount = 500

        ExpenseReport().printReport(listOf(expense))

        // will fail, because of timestamp in expense report
        Approvals.verify(output)
    }
}

