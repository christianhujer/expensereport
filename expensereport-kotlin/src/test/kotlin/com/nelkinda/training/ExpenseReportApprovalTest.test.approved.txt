Expenses Fri Jun 03 12:02:48 CEST 2022
Breakfast	500	 
Meal expenses: 500
Total expenses: 500
